.github/workflows/format.yml

```diff
@@ -34,22 +21,3 @@ jobs:
 
     - name: Run doctest
       run: make doctest
-
-  pre-commit-check:
```

Comment:
not very clear why we should remove this ? i find i useful since it somehow auto formats potential missing stuff

Reply:
Because pre-commit.ci already do this (and is already configured in this repo), this job is [deprecated](https://github.com/pre-commit/action)

Reply:
got it